src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,260 @@
+                IE_THROW() << "Inconsistency in RegistersPool::PhysicalSet::setAsUsed()";
+            }
+            unusedIndexes.erase(it);
+        }
+
+        void setAsUnused(int regIdx) {
+            assert(regIdx < size && regIdx >= 0);
+            unusedIndexes.insert(regIdx);
+        }
+
+        int getUnused() {
+            if (unusedIndexes.empty()) {
+                IE_THROW() << "Not enough registers in the RegistersPool";
+            }
+            return *unusedIndexes.begin();
+        }
+
+        void exclude(Xbyak::Reg reg) {
+            unusedIndexes.erase(reg.getIdx());
+        }
+
+        size_t countUnused() const {
+            return unusedIndexes.size();
+        }
+
+    private:
+        std::unordered_set<int> unusedIndexes;
+        int size;
+    };
+
+    virtual int getFreeOpmask() { IE_THROW() << "getFreeOpmask: The Opmask is not supported in current instruction set"; }
+    virtual void returnOpmaskToPool(int idx) { IE_THROW() << "returnOpmaskToPool: The Opmask is not supported in current instruction set"; }
+    virtual size_t countUnusedOpmask() const { IE_THROW() << "countUnusedOpmask: The Opmask is not supported in current instruction set"; }
+
+    virtual void excludeOpmask(const Xbyak::Opmask& reg) {
+    }
+
+    RegistersPool(std::initializer_list<Xbyak::Reg> regsToExclude, int simdRegistersNumber)
+            : simdSet(simdRegistersNumber) {
+        checkUniqueAndUpdate();
+        for (auto& reg : regsToExclude) {
+            if (reg.isXMM() || reg.isYMM() || reg.isZMM()) {
+                simdSet.exclude(reg);
+            } else if (reg.isOPMASK()) {
+                excludeOpmask(Xbyak::Opmask{reg.getIdx()});
+            } else if (reg.isREG()) {
+                generalSet.exclude(reg);
+            }
+        }
+        generalSet.exclude(Xbyak::Reg64(Xbyak::Operand::RSP));
+    }
+
+private:
+    template<typename TReg>
+    int getFree() {
+        static_assert(std::is_same<TReg, Xbyak::Xmm>::value || std::is_same<TReg, Xbyak::Ymm>::value || std::is_same<TReg, Xbyak::Zmm>::value ||
+                      std::is_same<TReg, Xbyak::Reg64>::value || std::is_same<TReg, Xbyak::Reg32>::value || std::is_same<TReg, Xbyak::Opmask>::value,
+                      "The AvxRegistersGuardPool::getFree() method unsupported type");
+        if (std::is_same<TReg, Xbyak::Xmm>::value || std::is_same<TReg, Xbyak::Ymm>::value || std::is_same<TReg, Xbyak::Zmm>::value) {
+            auto idx = simdSet.getUnused();
+            simdSet.setAsUsed(idx);
+            return idx;
+        } else if (std::is_same<TReg, Xbyak::Reg64>::value || std::is_same<TReg, Xbyak::Reg32>::value) {
+            auto idx = generalSet.getUnused();
+            generalSet.setAsUsed(idx);
+            return idx;
+        } else if (std::is_same<TReg, Xbyak::Opmask>::value) {
+            return getFreeOpmask();
+        }
+    }
+
+    template<typename TReg>
+    void returnToPool(TReg reg) {
```

Comment:
I think maybe better name is `free()` ?

Reply:
If `free()` is confusing we could use `release()` name

Reply:
Done